src/controllers/stallController.ts

```diff
@@ -25,9 +30,26 @@ async function retrieveStall(req: Request, res: Response, next: NextFunction) {
     const stall = await Stall.findByPk(req.params.id, {
       include: getStallInclude(),
     });
+
     if (stall === null) {
       throw new NotFoundError('Stall cannot be found');
     }
+
+    // To obtain single stall rating
+    const rating = (
+      await Review.findAll({
+        where: {
+          stallId: stall!.id,
+        },
+        attributes: [
+          [Sequelize.fn('ROUND', Sequelize.fn('AVG', Sequelize.col('rating')), 2), 'rating'],
+        ],
+      })
+    )[0].rating;
+
+    if (rating) stall.setDataValue('rating', rating);
```

Comment:
Can remove the conditional and just do `stall.setDataValue('rating', rating || 0);`

Reply:
Changed